lib/cisco_node_utils/vni.rb

```diff
@@ -43,55 +41,46 @@ def self.vnis
       hash
     end
 
-    def feature
-      if /N9K/.match(node.product_id)
-        vni = config_get('vni', 'feature_n9k')
-      else
-        vni = config_get('vni', 'feature')
-      end
-      return :disabled if vni.nil?
-      :enabled
+    # feature vni
+    def self.feature_vni_enabled
+      config_get('vni', 'feature')
+    rescue Cisco::CliError => e
+      # cmd will syntax reject when feature is not enabled
+      raise unless e.clierror =~ /Syntax error/
+      return false
     end
 
-    def feature_set(vni_set)
-      curr = feature
-      return if curr == vni_set
+    def self.feature_vni_enable
+      # feature nv overlay is a dependency for feature vni
+      Vni.feature_nv_overlay_enable unless Vni.feature_nv_overlay_enabled
```

Comment:
Well I'd say just remove the whole thing:
Vni.feature_nv_overlay_enable unless Vni.feature_nv_overlay_enabled. We don't need to enable nv overlay for vn-segment-vlan-based.

But another point is I have a test in minitest which tests feature on/off. In that the test will fail if nv overlay is also on be some other provider and we don't disable it before disabling vn-segment-vlan-based
I suggest we keep feature on/off test and the disablement part unless you feel otherwise.